wordpress/tools/media.py

```diff
@@ -0,0 +1,236 @@
+    suffix += f"&per_page={per_page}"
+    media_type = os.getenv("MEDIA_TYPE", "")
+    if media_type != "":
+        media_type_enum = ["image", "video", "text", "application", "audio"]
+        if media_type not in media_type_enum:
+            raise ValueError(
+                f"Error: Invalid media_type: {media_type}. media_type must be one of: {media_type_enum}."
+            )
+        suffix += f"&media_type={media_type}"
+    author_ids = os.getenv("AUTHOR_IDS", None)  # a list of comma separated author ids
+    if author_ids:
+        for author_id in author_ids.split(","):
+            if not author_id.isdigit():
+                raise ValueError(
+                    f"Error: Invalid author_ids: {author_id}. Author_ids must be a list of integers separated by commas."
+                )
+        suffix += f"&author={author_ids}"
+    search_query = os.getenv("SEARCH_QUERY", None)
+    if search_query:
+        suffix += f"&search={search_query}"
+
+    publish_after = os.getenv("PUBLISH_AFTER", None)
+    if publish_after:
+        if not is_valid_iso8601(publish_after):
+            raise ValueError(
+                f"Error: Invalid publish_after: {publish_after}. publish_after must be a valid ISO 8601 date string, in the format of YYYY-MM-DDTHH:MM:SS, or YYYY-MM-DDTHH:MM:SS+HH:MM."
+            )
+        suffix += f"&after={urllib.parse.quote(publish_after)}"
+    publish_before = os.getenv("PUBLISH_BEFORE", None)
+    if publish_before:
+        if not is_valid_iso8601(publish_before):
+            raise ValueError(
+                f"Error: Invalid publish_before: {publish_before}. publish_before must be a valid ISO 8601 date string, in the format of YYYY-MM-DDTHH:MM:SS, or YYYY-MM-DDTHH:MM:SS+HH:MM."
+            )
+        suffix += f"&before={urllib.parse.quote(publish_before)}"
+
+    modified_after = os.getenv("MODIFIED_AFTER", None)
+    if modified_after:
+        if not is_valid_iso8601(modified_after):
+            raise ValueError(
+                f"Error: Invalid modified_after: {modified_after}. modified_after must be a valid ISO 8601 date string, in the format of YYYY-MM-DDTHH:MM:SS, or YYYY-MM-DDTHH:MM:SS+HH:MM."
+            )
+        suffix += f"&modified_after={urllib.parse.quote(modified_after)}"
+
+    modified_before = os.getenv("MODIFIED_BEFORE", None)
+    if modified_before:
+        if not is_valid_iso8601(modified_before):
+            raise ValueError(
+                f"Error: Invalid modified_before: {modified_before}. modified_before must be a valid ISO 8601 date string, in the format of YYYY-MM-DDTHH:MM:SS, or YYYY-MM-DDTHH:MM:SS+HH:MM."
+            )
+        suffix += f"&modified_before={urllib.parse.quote(modified_before)}"
+    order = os.getenv("ORDER", "desc").lower()
+    order_enum = ["asc", "desc"]
+    if order not in order_enum:
+        raise ValueError(
+            f"Error: Invalid order: {order}. order must be one of: {order_enum}."
+        )
+    suffix += f"&order={order}"
```

Comment:
I think most of this stuff getting added to the suffix should be URL-encoded for safety.

Reply:
good point I will address those

Reply:
updated